app/controllers/aid_applications/finisheds_controller.rb

```diff
@@ -11,5 +11,15 @@ class FinishedsController < BaseController
     def edit
       @aid_application = current_aid_application
     end
+
+    def update
+      @aid_application = current_aid_application
+
+      if params[:form_action] == "resend_code"
+        if @aid_application.errors.empty?
```

Comment:
no errors here need to be checked because there isn't a save/validate action previously.

Reply:
done